core/src/main/java/org/infinispan/configuration/cache/TracingConfiguration.java

```diff
@@ -26,4 +30,33 @@ protected TracingConfiguration(AttributeSet attributes) {
    public boolean enabled() {
       return attributes.attribute(ENABLED).get();
    }
+
+   public boolean container() {
+      return attributes.attribute(CONTAINER).get();
+   }
+
+   public boolean cluster() {
+      return attributes.attribute(CLUSTER).get();
+   }
+
+   public boolean xSite() {
+      return attributes.attribute(X_SITE).get();
+   }
+
+   public boolean enabled(SpanCategory category) {
+      if (!enabled()) {
+         return false;
+      }
+
+      switch (category) {
+         case CONTAINER:
+            return container();
+         case CLUSTER:
+            return cluster();
+         case X_SITE:
+            return xSite();
+         default:
+            return true;
+      }
+   }
```

Comment:
My Intellij says it can be simplified as follow 
```suggestion
      return enabled() && switch (category) {
         case CONTAINER -> container();
         case CLUSTER -> cluster();
         case X_SITE -> xSite();
      };
```